src/functionaltests/java/com/ericsson/ei/subscriptions/query/QueryAggregatedObjectsTestSteps.java

```diff
@@ -0,0 +1,229 @@
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+        LOGGER.debug("Response of /queryAggregatedObject RestApi, Status Code: " + mvcResult.getResponse().getStatus() +
+                           "\nResponse: " + mvcResult.getResponse().getContentAsString());
+        
+        JsonNode jsonNodeResult = new ObjectMapper().readValue(mvcResult.getResponse().getContentAsString(), JsonNode.class);
+        JsonNode responseEntityNode = jsonNodeResult.get("responseEntity");
+        String responseEntityNodeFormatted = responseEntityNode.asText().replace("[", "").replace("]", "");
+        JsonNode responseEntityFormattedJsonNode = new ObjectMapper().readValue(responseEntityNodeFormatted, JsonNode.class);
+        
+        LOGGER.debug("AggregatedObject from Response: " + responseEntityFormattedJsonNode.toString());
+       
+        String actualTestCaseStartedEventId = responseEntityFormattedJsonNode.get("testCaseExecutions").get("testCaseStartedEventId").asText();
+        assertEquals(HttpStatus.OK.toString(), Integer.toString(mvcResult.getResponse().getStatus()));
+        assertEquals("Failed to compare actual Aggregated Object:\n" + expectedTestCaseStartedEventId
+                + "\nwith expected Aggregated Object:\n" + actualTestCaseStartedEventId,
+                expectedTestCaseStartedEventId, actualTestCaseStartedEventId);
+    }
+    
+    @And("^Perform an invalid query on same Aggregated object$")
+    public void perform_invalid_query_on_newly_created_aggregated_object() throws Throwable {
+        String invalidDocumentId = "5b3b4e1cccfbd55e6911dccc";
+        String entryPoint = "/queryAggregatedObject";
+        String expectedResponse = "{\"responseEntity\":\"[]\"}";
+        
+        LOGGER.debug("Trying an invalid query on /queryAggregatedObject RestApi with invalid documentId: " + invalidDocumentId);
+        mvcResult = mockMvc.perform(get(entryPoint)
+                .param("ID", invalidDocumentId)
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+        String responseAsString = mvcResult.getResponse().getContentAsString();
+        int reponseStatusCode = mvcResult.getResponse().getStatus();
+        LOGGER.debug("Response of /queryAggregatedObject RestApi, Status Code: " + reponseStatusCode +
+                "\nResponse: " + responseAsString);
+    
+        assertEquals(HttpStatus.OK.toString(), Integer.toString(reponseStatusCode));
+        assertEquals("Diffences between actual Aggregated Object:\n" + responseAsString
+                + "\nand expected Aggregated Object:\n" + expectedResponse,
+                expectedResponse, responseAsString);
+    }
+    
+    @And("^Perform a query for missed notification$")
+    public void perform_a_query_for_missed_notification() throws Throwable {
```

Comment:
This method is large, maybe break it out, also some formatting on long lines may be needed.